File: src/main/java/xyz/ptgms/excusemaker/ui/dashboard/DashboardFragment.java
package xyz.ptgms.excusemaker.ui.dashboard;

import android.content.ClipData;
import android.content.ClipboardManager;
import android.content.Context;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.CompoundButton;
import android.widget.Switch;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProviders;

import xyz.ptgms.excusemaker.MainActivity;
import xyz.ptgms.excusemaker.R;
import xyz.ptgms.excusemaker.ui.home.HomeFragment;

public class DashboardFragment extends Fragment {
    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_dashboard, container, false);

        MainActivity activity = (MainActivity) getActivity();
        boolean lowercaseState = activity.LoadSetting(0);
        boolean uppercaseState = activity.LoadSetting(1);
        boolean furryState = activity.LoadSetting(2);
        boolean mockeryState = activity.LoadSetting(3);

        final Switch lowercase = view.findViewById(R.id.switch1);
        final Switch uppercase = view.findViewById(R.id.switch2);
        Switch furry = view.findViewById(R.id.switch3);
        final Switch mockery = view.findViewById(R.id.switch4);
        Button source = view.findViewById(R.id.sourceButton);

        lowercase.setChecked(lowercaseState);
        uppercase.setChecked(uppercaseState);
        furry.setChecked(furryState);
        mockery.setChecked(mockeryState);

        lowercase.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener()
        {
            @Override
            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                MainActivity activity = (MainActivity) getActivity();
                activity.saveSetting(0, isChecked);
                mockery.setChecked(false);
                if (uppercase.isChecked() && isChecked) {
                    uppercase.setChecked(false);
                    activity.saveSetting(1, false);
                }
                if (mockery.isChecked() && isChecked) {
                    mockery.setChecked(false);
                    activity.saveSetting(3, false);
                }

            }
        });
        uppercase.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener()
        {
            @Override
            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                MainActivity activity = (MainActivity) getActivity();
                activity.saveSetting(1, isChecked);
                if (lowercase.isChecked() && isChecked) {
                    lowercase.setChecked(false);
                    activity.saveSetting(0, false);
                }
                if (mockery.isChecked() && isChecked) {
                    mockery.setChecked(false);
                    activity.saveSetting(3, false);
                }

            }
        });
        furry.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener()
        {
            @Override
            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                MainActivity activity = (MainActivity) getActivity();
                activity.saveSetting(2, isChecked);

            }
        });
        mockery.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener()
        {
            @Override
            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                MainActivity activity = (MainActivity) getActivity();
                activity.saveSetting(3, isChecked);
                if (uppercase.isChecked() && isChecked) {
                    uppercase.setChecked(false);
                    activity.saveSetting(0, false);
                }
                if (lowercase.isChecked() && isChecked) {
                    lowercase.setChecked(false);
                    activity.saveSetting(1, false);
                }

            }
        });
        source.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String url = "https://github.com/ptgms/ExcuseMaker-Android";
                Intent i = new Intent(Intent.ACTION_VIEW);
                i.setData(Uri.parse(url));
                startActivity(i);
            }
        });
        return view;
    }
}

File: app/src/main/java/xyz/ptgms/excusemaker/ui/dashboard/DashboardFragment.kt
package xyz.ptgms.excusemaker.ui.dashboard

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.Switch
import androidx.fragment.app.Fragment
import xyz.ptgms.excusemaker.MainActivity
import xyz.ptgms.excusemaker.R

class DashboardFragment : Fragment() {
    override fun onCreateView(inflater: LayoutInflater,
                              container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_dashboard, container, false)
        val activity = activity as MainActivity?
        val lowercaseState = activity!!.LoadSetting(0)
        val uppercaseState = activity.LoadSetting(1)
        val furryState = activity.LoadSetting(2)
        val mockeryState = activity.LoadSetting(3)
        val lowercase = view.findViewById<Switch>(R.id.switch1)
        val uppercase = view.findViewById<Switch>(R.id.switch2)
        val furry = view.findViewById<Switch>(R.id.switch3)
        val mockery = view.findViewById<Switch>(R.id.switch4)
        val source = view.findViewById<Button>(R.id.sourceButton)
        lowercase.isChecked = lowercaseState
        uppercase.isChecked = uppercaseState
        furry.isChecked = furryState
        mockery.isChecked = mockeryState
        lowercase.setOnCheckedChangeListener { buttonView, isChecked ->
            val activity = getActivity() as MainActivity?
            activity!!.saveSetting(0, isChecked)
            mockery.isChecked = false
            if (uppercase.isChecked && isChecked) {
                uppercase.isChecked = false
                activity.saveSetting(1, false)
            }
            if (mockery.isChecked && isChecked) {
                mockery.isChecked = false
                activity.saveSetting(3, false)
            }
        }
        uppercase.setOnCheckedChangeListener { buttonView, isChecked ->
            val activity = getActivity() as MainActivity?
            activity!!.saveSetting(1, isChecked)
            if (lowercase.isChecked && isChecked) {
                lowercase.isChecked = false
                activity.saveSetting(0, false)
            }
            if (mockery.isChecked && isChecked) {
                mockery.isChecked = false
                activity.saveSetting(3, false)
            }
        }
        furry.setOnCheckedChangeListener { buttonView, isChecked ->
            val activity = getActivity() as MainActivity?
            activity!!.saveSetting(2, isChecked)
        }
        mockery.setOnCheckedChangeListener { buttonView, isChecked ->
            val activity = getActivity() as MainActivity?
            activity!!.saveSetting(3, isChecked)
            if (uppercase.isChecked && isChecked) {
                uppercase.isChecked = false
                activity.saveSetting(0, false)
            }
            if (lowercase.isChecked && isChecked) {
                lowercase.isChecked = false
                activity.saveSetting(1, false)
            }
        }
        source.setOnClickListener {
            val url = "https://github.com/ptgms/ExcuseMaker-Android"
            val i = Intent(Intent.ACTION_VIEW)
            i.data = Uri.parse(url)
            startActivity(i)
        }
        return view
    }
}